qemu/tests/rng_maxbytes_period.py

```diff
@@ -0,0 +1,85 @@
+import re
+import logging
+
+from virttest import error_context
+from virttest import env_process
+from virttest import virt_vm
+from virttest import utils_misc
+
+from aexpect.exceptions import ShellTimeoutError
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu virtio-rng device test:
+    1) boot guest with virtio-rng device
+    2) read random data in guest
+    3) check the read data rate
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _is_rngd_running():
+        """
+        Check whether rngd is running
+        """
+        output = session.cmd_output(check_rngd_service)
+        if 'running' not in output:
+            return False
+        return True
+
+    timeout = float(params.get("login_timeout", 360))
+    read_rng_timeout = float(params.get("read_rng_timeout", 3600))
+    read_rng_cmd = params["read_rng_cmd"]
+    max_bytes = params.get("max-bytes_virtio-rng-pci")
+    period = params.get("period_virtio-rng-pci")
+
+    if not max_bytes and not period:
+        test.error("Please specify the expected max-bytes and/or period.")
+    if not max_bytes or not period:
+        if max_bytes != '0':
+            error_info = params["expected_error_info"]
+            try:
+                env_process.process(test, params, env,
+                                    env_process.preprocess_image,
+                                    env_process.preprocess_vm)
+            except virt_vm.VMCreateError as e:
```

Comment:
I am afraid we cannot mix the positive test and negative test here, please consider the following situation:
what will happen if the positive test triggers an exception here? 
for the positive test, we call env_process_process(), and expect no issue, so we need to test  if any exception will be caught, and test always failed if any exception is caught. Here, if error_info not in e, the test passed...

also, we can use str(e) here

Reply:
We can divide the test into two parts:
do_positive_test() 
do_negative_test()
In positive test function, do the test and check the rate; while in negative test function, do the test and check the proper error message(for ShellTimeout exception, we can still check an error msg (expected_error_info = timeout xxxx), which can keep the same check with other negative settings

just FYI.